packages/components/src/unit-control/index.tsx

```diff
@@ -92,24 +106,24 @@ function UnitControl(
 		setUnit( next );
 	};
 
-	const mayUpdateUnit = ( event ) => {
-		if ( ! isNaN( event.target.value ) ) {
+	const mayUpdateUnit = ( event: SyntheticEvent< HTMLInputElement > ) => {
+		if ( ! isNaN( Number( event.currentTarget.value ) ) ) {
 			refParsedValue.current = null;
 			return;
 		}
 		const [ parsedValue, parsedUnit ] = getValidParsedUnit(
-			event.target.value,
+			event.currentTarget.value,
```

Comment:
Why was this change necessary? Are we confident that it won't introduce a different behaviour of the component?